src/service/apiClient.js

```diff
@@ -1,59 +1,61 @@
-import { API_URL } from "./constants"
+import { API_URL } from "./constants";
 
 async function login(email, password) {
-    return await post('login', { email, password }, false)
+  return await post("login", { email, password }, false);
 }
 
 async function register(email, password) {
-    await post('users', { email, password }, false)
-    return await login(email, password)
+  const res = await post("users", { email, password }, false);
+
+  if (res.data.error) return res;
```

Comment:
Let's stick with `{ }` code blocks to keep things readable and consistent with the rest of the codebase